File: checkvest/ios/Sources/CheckVestPlugin/CheckVestPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CheckVestPlugin)
public class CheckVestPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CheckVestPlugin"
    public let jsName = "CheckVest"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "checkHasVest", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = CheckVest()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    @objc func checkHasVest(_ call: CAPPluginCall) {
        let imageBase64 = call.getString("imageBase64") ?? ""
        let showLogs = call.getBool("showLogs") ?? false
        
        let hasVest = implementation.checkHasVest(imageBase64: imageBase64, showLogs: showLogs)
        
        call.resolve([
            "hasVest": hasVest
        ])
    }
}

File: checkvest/android/src/main/java/com/odc/plugins/checkvest/CheckVestPlugin.java
package com.odc.plugins.checkvest;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.annotation.CapacitorPlugin;
//import com.getcapacitor.annotation.PluginMethod;
import com.getcapacitor.JSObject;

@CapacitorPlugin(name = "CheckVest")
public class CheckVestPlugin extends Plugin {
    private CheckVest implementation;

    @Override
    public void load() {
        implementation = new CheckVest(getContext());
    }




    //@PluginMethod
    public void checkHasVest(PluginCall call) {
        String base64 = call.getString("imageBase64");
        boolean showLogs = Boolean.TRUE.equals(call.getBoolean("showLogs", false));

        boolean hasVest = implementation.checkHasVest(base64, showLogs);

        JSObject ret = new JSObject();
        ret.put("hasVest", hasVest);
        call.resolve(ret);
    }
}